### ISSUE TITLE
Feature: gitoxide

### ISSUE THREAD

**Issue by martintc** [comtrya/comtrya#347]
**Description**

Mostly as a research/possible implementation case for me, along with notes. It looks like gitoxide may have enough functionality to finally replace git2-rs. Git2-rs is our main offender with a dependency on openssl which has been a thorn in the side. Gitoxide uses rustls as a backend.

**Additional information**

Research to ensure functionality is present. Replace git2-rs with gitoxide if possible.

**Question to the community**

Is there a reason why we should maintain git2-rs support and perhaps place git2-rs vs gitoxide behind a feature flag?

**Comment by martintc** [comment]
Unfortunately at this time, gix is still missing large parts of functionality we would need to upstream gix/gitoxide support into one of our dependencies.

**Comment by yonas** [comment]
> gix is still missing large parts of functionality we would need to upstream gix/gitoxide support into one of our dependencies.

@martintc What features does comtrya need that are missing in gitoxide?

**Comment by rawkode** [comment]
It can't perform the equivilent of a `git pull`, which is a fetch, merge, and potentially rebase.

IIRC, merge and rebase aren't implemented and it's been that way for a while.

**Comment by martintc** [comment]
I believe it also had an issue with pulling releases also. Another, if I remember was being able to use ssh keys or something along those lines to pull from private git repositories. This was also several months ago and I forgot most of it. Essentially, at the bare minimum, we need to replace  Rawkode's gitsync crate or expose using gitoxide in lieu of git2 via a feature flag in gitsync. I did quiet a bit of the work for getting interfaces aligned and using feature flags in gitaync, but once I got to actually implementing the gitoxide side, it didn't have enough functionality at the time to do it.
